core/unit_test/TestPageMigration.hpp

```diff
@@ -0,0 +1,242 @@
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#if defined _WIN32  // windows system
+#include <windows.h>
+unsigned getBytesPerPage() {
+  SYSTEM_INFO si;
+  GetSystemInfo(&si);
+  return si.dwPageSize;
+}
+#else  // unix/posix system
+#include <unistd.h>
+unsigned getBytesPerPage() { return sysconf(_SC_PAGESIZE); }
+#endif
+
+#include <Kokkos_Core.hpp>
+
+#include <algorithm>
+#include <numeric>
+#include <iostream>
+
+namespace {
+void printTimings(std::ostream& out, std::vector<double> const& tr) {
+  out << "TimingResult contains " << tr.size() << " results:\n";
+  for (auto it = tr.begin(); it != tr.end(); ++it) {
+    out << "Duration of loop " << it - tr.begin() << " is " << *it
+        << " seconds\n";
+  }
+}
+
+template <typename T>
+T computeMean(std::vector<T> const& results) {
+  return std::accumulate(results.begin(), results.end(), T{}) / results.size();
+}
+
+template <typename ExecSpace, typename ViewType>
+std::vector<double> incrementInLoop(ViewType& view,
+                                    unsigned int numRepetitions) {
+  Kokkos::Timer timer;
+  std::vector<double> results;
+
+  Kokkos::fence();
+  for (unsigned i = 0; i < numRepetitions; ++i) {
+    timer.reset();
+    Kokkos::parallel_for(
+        "increment",
+        Kokkos::RangePolicy<ExecSpace,
+                            Kokkos::IndexType<decltype(view.size())>>{
+            0, view.size()},
+        KOKKOS_LAMBDA(auto i) { ++view(i); });
```

Comment:
Does `nvcc` really allow using generic `__host__ __device__` lambdas? I wouldn't be surprised if using `i` here raises a shadowing warning on some compilers.

Reply:
you are right -> changed in [311f08](https://github.com/kokkos/kokkos/pull/5289/commits/311f0811365c3875bcd1bc30241c8536fbcd3f22)